src/transformers/generation_utils.py

```diff
@@ -359,12 +359,69 @@ class BeamSampleEncoderDecoderOutput(ModelOutput):
 BeamSampleOutput = Union[BeamSampleEncoderDecoderOutput, BeamSampleDecoderOnlyOutput]
 
 
+ENCODER_MODEL_INPUT_NAMES = ["input_ids", "inputs_embeds", "input_values", "input_features", "pixel_values"]
+
+
 class GenerationMixin:
     """
     A class containing all of the functions supporting generation, to be used as a mixin in
     :class:`~transformers.PreTrainedModel`.
     """
 
+    def _prepare_model_inputs(
```

Comment:
This function extracts the correct input & input_model_name depending on well-explained scenarios